ref/src/sw_driver.F90

```diff
@@ -97,6 +120,22 @@ program sw_driver
   ! Read the input state from the NetCDF input file
   call read_state(TRIM(input_data_dir) // '/' // TRIM(input_file))
 
+#ifdef ENABLE_MPI
+  ! Make sure rows*cols equal the total nmber of MPI tasks
+  if (rows*cols /= nTasks) then
+    print*,'Error in sw_driver running in parallel mode'
+    print*,'rows*cols must equal the number of MPI tasks'
+    print*,'rank,rows,cols,nTasks=', rank,rows,cols,nTasks
+    stop
+  endif
+
+  ! Get the MPI task neighbore for the exchange
+  call get_neighbors
+
+  ! Allocate statuses for MPI_WAITALL: 0:15 because there are 8 sends and 8 receives
+  allocate( statuses(MPI_STATUS_SIZE, 0:15) )
```

Comment:
Is there a better place for this?  Having it here seems out of place.

Reply:
@christopherwharrop-noaa I could put the allocate in with the other ollocates but I would need to enclose it in and IFDEF on MPI.